Database/SQLite/Simple.hs

```diff
@@ -482,6 +483,22 @@ convertRow fromRow_ rowRes ncols = do
       where
         bs = T.pack $ show sql
 
+-- | Run an IO action inside a SQL transaction started with @BEGIN
+-- IMMEDIATE TRANSACTION@.  If the action throws any kind of an exception, the
+-- transaction will be rolled back with @ROLLBACK TRANSACTION@.
+-- Otherwise the results are committed with @COMMIT TRANSACTION@.
+withImmediateTransaction :: Connection -> IO a -> IO a
+withImmediateTransaction conn action =
```

Comment:
can you please expand in your docs a bit why this is useful?  not everyone knows what immediate transactions are, even though they might need them.

also, maybe add a generalized withTransactionPrivate that can do both immediate and normal, and then make withTransaction and withImmediateTransaction use that instead of duplicating?